src/extensions/mod.rs

```diff
@@ -228,7 +228,7 @@ pub trait Extension: Debug + ExtensionHelper {
     /// Get a reference to the `RatchetTreeExtension`.
     /// Returns an `InvalidExtensionType` error if called on an `Extension`
     /// that's not a `RatchetTreeExtension`.
-    fn to_ratchet_tree_extension_ref(&self) -> Result<&RatchetTreeExtension, ExtensionError> {
```

Comment:
Wouldn't this be `as_ratchet_tree_extension`, as it's just a cast of a reference?